docs/design/design.md

```diff
@@ -49,11 +49,52 @@ This means we aim to standardize:
      - `kube-scheduler.yaml`
  - `/etc/kubernetes/kubelet.conf` as the path where the kubelet should store its credentials to the API server.
  - `/etc/kubernetes/admin.conf` as the path from where the admin can fetch his/her superuser credentials.
+ - Names of certificates files:
+   - `ca.crt`, `ca.key`
+   - `apiserver.crt`, `apiserver.key`
+   - `apiserver-kubelet-client.crt`, `apiserver-kubelet-client.key`
+   - `sa.pub`, `sa.key`
+   - `front-proxy-ca.crt`, `front-proxy-ca.key`
+   - `front-proxy-client.crt`, `front-proxy-client.key`
 
 
 ## `kubeadm init` phases
 
-### Phase 1: Generate the necessary certificates
+`kubeadm init` internal workflow consists of a sequence of phases.
+
+The `Kubeadm phase` command, alpha in 1.8, allows users to invoke individually each phase, and ultimately offers a reusable and composable API/toolbox that can be used by other kubernetes bootstrap tools / by any IT automation tool / by advanced user for creating custom clusters. 
+
+### Preflight checks
+
+`kubeadm` executes a set of preflight checks before starting the init, with the aim to verify preconditions and avoid common cluster startup problems. 
+In any case the user can skip preflight checks with the `--skip-preflight-checks` option.
+
+- [warning] if kubeadm version "not safe" for installing the required kubernetes version (kubeadm is considered safe for installing patch versions "newest" than kubeadm version itself, not safe for installing "newest" major/minor releases)
+- Kubernetes system requirements, 
+  - [error] if not linux, 
+  - [error] if not Kernel 3.10+ or 4+ with specific KernelSpec,
+  - [error] if not CGroups (with a specific set of subsystems) in place,
+  - [error/warning] if Docker endpoint does not exists, if docker version >1.9, <= 1.12,
```

Comment:
It won't error if docker isn't present I think. It will error if docker is present but not working.
The version range for v1.8 is `v1.11.2 <= x <= v1.13.1`

Reply:
Done, but I'm not sure the code is aligned with this... 
`system.DefaultSysSpec` contains this regular expression 
`Version: []string{`1\.(9|1[0-2])\..*`}, // Requires 1.9+``

Reply:
DefaultSysSpec at HEAD:
```go
var DefaultSysSpec = SysSpec{
	OS: "Linux",
	KernelSpec: KernelSpec{
		Versions: []string{`3\.[1-9][0-9].*`, `4\..*`}, // Requires 3.10+ or 4+
		// TODO(random-liu): Add more config
		// TODO(random-liu): Add description for each kernel configuration:
		Required: []KernelConfig{
			{Name: "NAMESPACES"},
			{Name: "NET_NS"},
			{Name: "PID_NS"},
			{Name: "IPC_NS"},
			{Name: "UTS_NS"},
			{Name: "CGROUPS"},
			{Name: "CGROUP_CPUACCT"},
			{Name: "CGROUP_DEVICE"},
			{Name: "CGROUP_FREEZER"},
			{Name: "CGROUP_SCHED"},
			{Name: "CPUSETS"},
			{Name: "MEMCG"},
			{Name: "INET"},
			{Name: "EXT4_FS"},
			{Name: "PROC_FS"},
			{Name: "NETFILTER_XT_TARGET_REDIRECT", Aliases: []string{"IP_NF_TARGET_REDIRECT"}},
			{Name: "NETFILTER_XT_MATCH_COMMENT"},
		},
		Optional: []KernelConfig{
			{Name: "OVERLAY_FS", Aliases: []string{"OVERLAYFS_FS"}, Description: "Required for overlayfs."},
			{Name: "AUFS_FS", Description: "Required for aufs."},
			{Name: "BLK_DEV_DM", Description: "Required for devicemapper."},
		},
		Forbidden: []KernelConfig{},
	},
	Cgroups: []string{"cpu", "cpuacct", "cpuset", "devices", "freezer", "memory"},
	RuntimeSpec: RuntimeSpec{
		DockerSpec: &DockerSpec{
			Version:     []string{`1\.1[1-3]\..*`}, // Requires 1.11+
			GraphDriver: []string{"aufs", "overlay", "overlay2", "devicemapper"},
		},
	},
}
```